plothist/plotters.py

```diff
@@ -469,107 +321,237 @@ def plot_error_hist(hist, ax, **kwargs):
 
     Parameters
     ----------
-    hist_1, hist_2 : boost_histogram.Histogram
-        The histograms to be compared.
+    hist_1 : boost_histogram.Histogram
+        The first histogram to compare.
+    hist_2 : boost_histogram.Histogram
+        The second histogram to compare.
     xlabel : str, optional
-        The label for the x-axis of the comparison plot.
+        The label for the x-axis. Default is None.
     ylabel : str, optional
-        The label for the y-axis of the comparison plot.
-    x1_label, x2_label : str, optional
-        The labels for the two histograms being compared.
-    save_as : str, optional
-        If provided, the filename to save the figure as.
+        The label for the y-axis. Default is None.
+    x1_label : str, optional
+        The label for the first histogram. Default is "x1".
+    x2_label : str, optional
+        The label for the second histogram. Default is "x2".
+    comparison : str, optional
+        The type of comparison to plot. Default is "ratio".
+    comparison_ylim : tuple or None, optional
+        The y-axis limits for the comparison plot. Default is None.
+    save_as : str or None, optional
+        The path to save the figure. Default is None.
+    fig : matplotlib.figure.Figure or None, optional
+        The figure to use for the plot. If fig, ax_main and ax_comparison are None, a new figure will be created. Default is None.
+    ax_main : matplotlib.axes.Axes or None, optional
+        The main axes for the histogram comparison. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
+    ax_comparison : matplotlib.axes.Axes or None, optional
+        The axes for the comparison plot. If fig, ax_main and ax_comparison are None, a new axes will be created. Default is None.
+    **kwargs
+        Additional keyword arguments to be passed to the plot_comparison function.
 
     Returns
     -------
     fig : matplotlib.figure.Figure
-        The generated figure.
+        The created figure.
+    ax_main : matplotlib.axes.Axes
+        The main axes for the histogram comparison.
     ax_comparison : matplotlib.axes.Axes
-        The axes for the histogram comparison plot.
-    ax_ratio : matplotlib.axes.Axes
-        The axes for the ratio plot.
+        The axes for the comparison plot.
+
+    See Also
+    --------
+    plot_comparison : Plot the comparison between two histograms.
 
     """
     if not np.all(hist_1.axes[0].edges == hist_2.axes[0].edges):
         raise ValueError("The bins of the compared histograms must be equal.")
 
-    fig, (ax_comparison, ax_ratio) = plt.subplots(
-        2, gridspec_kw={"height_ratios": [4, 1]}
-    )
+    if fig is None and ax_main is None and ax_comparison is None:
+        fig, (ax_main, ax_comparison) = create_comparison_figure()
+    elif (fig is not None or ax_main is not None or ax_comparison is not None) and (
+        fig is None or ax_main is None or ax_comparison is None
```

Comment:
At this stage, we already know that at least one is not `None`, so just need to check if at least one is `None`, and raise the Error in this case